src/category_theory/isomorphism.lean

```diff
@@ -123,8 +123,44 @@ instance of_iso (f : X ≅ Y) : is_iso f.hom :=
 instance of_iso_inverse (f : X ≅ Y) : is_iso f.inv :=
 { inv := f.hom }
 
+variables {f g : X ⟶ Y} {h : Y ⟶ Z}
+
+instance inv_is_iso [is_iso f] : is_iso (category_theory.inv f) :=
+{ inv := f,
+  hom_inv_id' := inv_hom_id f,
+  inv_hom_id' := hom_inv_id f }
+instance comp_is_iso [is_iso f] [is_iso h] : is_iso (f ≫ h) :=
+{ inv := category_theory.inv h ≫ category_theory.inv f,
+  hom_inv_id' := begin erw [category.assoc, hom_inv_id_assoc], exact hom_inv_id f, end,
+  inv_hom_id' := begin erw [category.assoc, inv_hom_id_assoc], exact inv_hom_id h, end }
+
+@[simp] lemma inv_id : category_theory.inv (𝟙 X) = 𝟙 X := rfl
+@[simp] lemma inv_comp [is_iso f] [is_iso h] :
+  category_theory.inv (f ≫ h) = category_theory.inv h ≫ category_theory.inv f := rfl
+@[simp] lemma is_iso.inv_inv [is_iso f] : category_theory.inv (category_theory.inv f) = f := rfl
+@[simp] lemma iso.inv_inv (f : X ≅ Y) :
+  category_theory.inv (f.inv) = f.hom := rfl
+@[simp] lemma iso.inv_hom (f : X ≅ Y) :
+  category_theory.inv (f.hom) = f.inv := rfl
+
```

Comment:
Do you also want two lemmas about composing `f` with it's inverse (for `is_iso`)?

Reply:
Isn't that already `is_iso.inv_hom_id`/`is_iso.hom_inv_id`?

Reply:
Ah, of course... me being silly.